debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -126,15 +128,33 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                         dispatcher.dispatchHeartbeatEvent(offsetContext);
                         return;
                     }
+                    if (TransactionMessage.isTransactionalMessage(message) && !connectorConfig.shouldProvideTransactionMetadata()) {
+                        LOGGER.trace("Received transactional message {}", message);
+                        return;
```

Comment:
Isn't the early return a problem as we'd miss the update of `lastCompletelyProcessedLsn` below?